wpimath/src/test/native/cpp/geometry/Pose2dTest.cpp

```diff
@@ -54,6 +55,94 @@ TEST(Pose2dTest, Minus) {
   EXPECT_DOUBLE_EQ(0.0, transform.Rotation().Degrees().value());
 }
 
+TEST(Pose2dTest, Nearest) {
+  const Pose2d origin{0_m, 0_m, 0_deg};
+
+  const Pose2d pose1{
+      Translation2d{1_m, Rotation2d{units::degree_t{
+                             static_cast<double>(std::rand() % 360)}}},
+      0_deg};
+  const Pose2d pose2{
+      Translation2d{2_m, Rotation2d{units::degree_t{
+                             static_cast<double>(std::rand() % 360)}}},
+      0_deg};
+  const Pose2d pose3{
+      Translation2d{3_m, Rotation2d{units::degree_t{
+                             static_cast<double>(std::rand() % 360)}}},
+      0_deg};
+  const Pose2d pose4{
+      Translation2d{4_m, Rotation2d{units::degree_t{
+                             static_cast<double>(std::rand() % 360)}}},
+      0_deg};
+  const Pose2d pose5{
+      Translation2d{5_m, Rotation2d{units::degree_t{
+                             static_cast<double>(std::rand() % 360)}}},
+      0_deg};
+
+  EXPECT_DOUBLE_EQ(
```

Comment:
Perhaps use `EXPECT_EQ` on the `Translation2d`s directly?

Reply:
Doing so does not output good information for debugging, instead printing the binary representation:
```
error: Expected equality of these values:
  translation1
    Which is: 16-byte object <00-00 00-00 00-00 F0-3F 00-00 00-00 00-00 00-40>
  translation2
    Which is: 16-byte object <00-00 00-00 00-00 00-40 00-00 00-00 00-00 F0-3F>
```
All of the geometry objects test like this.

https://github.com/wpilibsuite/allwpilib/pull/4675#issuecomment-1321404941

We could fix this as mentioned in the above conversation but that's beyond the scope of this PR.

I'm going to change the `EXPECT_EQ` to `EXPECT_DOUBLE_EQ` like they should be, though.